src/services/edusharing/logic/connector.js

```diff
@@ -0,0 +1,165 @@
+const request = require('request-promise-native');
+const REQUEST_TIMEOUT = 8000; // ms
+
+// STACKOVERFLOW BEAUTY
+function validURL(str) {
+	const pattern = new RegExp(
+		'^(https?:\\/\\/)?' +
+		'((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' +
+		'((\\d{1,3}\\.){3}\\d{1,3}))' +
+		'(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' +
+		'(\\?[;&a-z\\d%_.~+=-]*)?' +
+		'(\\#[-a-z\\d_]*)?$',
+		'i'
+	);
+	return !!pattern.test(str);
+}
+
+class EduSharingConnector {
+	constructor() {
+		if (EduSharingConnector.instance) {
+			return EduSharingConnector.instance;
+		}
+		if (!validURL(this.url)) {
+			return 'Invalid ES_DOMAIN, check your .env';
+		}
+		this.authorization = null; // JSESSION COOKIE
+		this.accessToken = null; // ACCESSTOKEN
+		EduSharingConnector.instance = this;
+	}
+
+	static get headers() {
+		return {
+			Accept: 'application/json',
+			'Content-type': 'application/json'
+		};
+	}
+
+	static get authorization() {
+		const userName = process.env.ES_USER;
+		const pw = process.env.ES_PASSWORD;
+		const headers = Object.assign({}, EduSharingConnector.headers, {
+			Authorization: `Basic ${Buffer.from(`${userName}:${pw}`).toString(
+				'base64'
+			)}`
+		});
+		return headers;
+	}
+
+	// gets cookie (JSESSION) and attach it to header
+	getCookie() {
+		const cookieOptions = {
+			uri: `${process.env.ES_DOMAIN}/edu-sharing/rest/authentication/v1/validateSession`,
+			method: 'GET',
+			headers: EduSharingConnector.authorization,
+			resolveWithFullResponse: true,
+			json: true
+		};
+		return request(cookieOptions)
+			.then(result => {
+				if (
+					result.statusCode !== 200 ||
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['||' should be placed at the beginning of the line.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4791331)